code/queue.h

```diff
@@ -21,6 +24,90 @@ class CQueue
 	static void		AddToQueue(int nTimer, int nArgs, ...);
 	static bool		HasQueuePending(void *qChar);	
 	static void		DeleteQueuedEventsInvolving(void *qChar);
+
+	/// Adds a function pointer to the queue with related arguments. The function will be executed at the specified time (in ticks).
+	/// @tparam Func: Template type of the function to call. 
+	/// @tparam ...Args: Template parameter pack used to specify the variadic arguments. 
+	/// @param nTimer: The time (in ticks) which the function executes.
+	/// @param func: The function to execute at the specified time.
+	/// @param ...args: Variadic arguments used to call the function.
+	template<class Func, class... Args>
+	void AddToNewQueue(int nTimer, Func func, Args &&...args)
```

Comment:
Is the `New` just to signify that this is the new way of handling it?

Reply:
nvm, I see your notes now. I was just a little confused at first as to whether this was supposed to be creating a _new_ queue and then adding to it.